### PR TITLE
doc: Add hint towards using ArgumentParser.add_argument_group

### PR DESCRIPTION
**Pull request by EricCousineau-TRI**
Since pl adds many arguments, it is nice to distinguish these arguments

## What does this PR do?

Adds and tests a workflow of using `ArgumentParser.add_argument_group`:
https://docs.python.org/3.2/library/argparse.html#argparse.ArgumentParser.add_argument_group

- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- (N/A, I think?) Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [ ] **Check that target branch and milestone match!**

## Did you have fun?

Yuhhhhhhhhh :parachute:

### CODE REVIEW

**Comment by Borda** [comment]
@EricCousineau-TRI thank you for sending your PR, just a minor issue coming from our side... we have swapped branches regarding upcoming feat 1.2, mind rebase on actual `master`? if needed see [How to fix PR with mixed base and target branches?](https://pytorch-lightning.readthedocs.io/en/latest/generated/CONTRIBUTING.html#how-to-fix-pr-with-mixed-base-and-target-branches)

**Comment by EricCousineau-TRI** [comment]
Done!

File: docs/source/common/trainer.rst [hunk 1]
```diff
@@ -128,7 +128,9 @@ So you can run it like so:
 
     if __name__ == '__main__':
         parser = ArgumentParser()
-        parser = Trainer.add_argparse_args(parser)
+        parser = Trainer.add_argparse_args(
+            parser.add_argument_group(title="pl.Trainer args")
```
Comment (carmocca) [inline docs/source/common/trainer.rst:132]:
```suggestion
            # Group the Trainer arguments together
            parser.add_argument_group(title="pl.Trainer args")
```

**Review by carmocca — APPROVED** [review]

File: docs/source/common/trainer.rst (hunk 1 above)
Comment (EricCousineau-TRI) [reply docs/source/common/trainer.rst:133]: Done in 5526b42d. Changed casing to match other comments in the file (e.g. `# print full summary of all modules and submodules`)

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Neat addition !

**Review by Borda — APPROVED** [review]

**Pull request merged**